api/v1/service/resource/resource.go

```diff
@@ -176,6 +178,52 @@ func (s *service) ByCatalogKindNameVersion(ctx context.Context, p *resource.ByCa
 	}
 }
 
+func (s *service) fetchCatalogClonePath(ctx context.Context, catalogName string) (string, error) {
```

Comment:
Could you please move this function to `shared/resource.go` since all db transactions are maintained in that file